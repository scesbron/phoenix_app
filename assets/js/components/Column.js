import React from "react";
import PropTypes from "prop-types";

export default class Column extends React.Component {
  render() {
    const { isDraggingOver, title, children } = this.props;
    return (
      <div className="column">
        <div className="column-header">
          <div className="column-header-title">{title}</div>
        </div>
        <div className={`column-content ${isDraggingOver ? "column-content-over" : ""}`}>
          {children}
        </div>
      </div>
    );
  }
}

Column.propTypes = {
  isDraggingOver: PropTypes.bool,
  title: PropTypes.string.isRequired,
  children: PropTypes.node
};
