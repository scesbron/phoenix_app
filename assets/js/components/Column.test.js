import React from "react";
import Column from "./Column";
import renderer from "react-test-renderer";

it("renders correctly on default state", () => {
  const tree = renderer
    .create(
      <Column title="Title">
        <div>Content</div>
      </Column>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when dragging is over", () => {
  const tree = renderer
    .create(
      <Column title="Title" isDraggingOver={true}>
        <div>Content</div>
      </Column>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
