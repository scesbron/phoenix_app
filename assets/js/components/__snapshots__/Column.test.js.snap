// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly on default state 1`] = `
<div
  className="column"
>
  <div
    className="column-header"
  >
    <div
      className="column-header-title"
    >
      Title
    </div>
  </div>
  <div
    className="column-content "
  >
    <div>
      Content
    </div>
  </div>
</div>
`;

exports[`renders correctly when dragging is over 1`] = `
<div
  className="column"
>
  <div
    className="column-header"
  >
    <div
      className="column-header-title"
    >
      Title
    </div>
  </div>
  <div
    className="column-content column-content-over"
  >
    <div>
      Content
    </div>
  </div>
</div>
`;
